enverus_direct_access/__init__.py

```diff
@@ -523,3 +442,154 @@ def query(self, dataset, **options):
+        :type client_id: str
+        :param client_secret: client secret credential.
+        :type client_secret: str
+        :param api_key: api key credential.
+        :type api_key: str
+        :param retries: the number of attempts when retrying failed requests with status codes of 500, 502, 503 or 504
+        :type retries: int
+        :param backoff_factor: the factor to use when exponentially backing off prior to retrying a failed request
+        :type backoff_factor: int
+        :param links: a dictionary of prev and next links as provided by the python-requests Session.
+        See https://requests.readthedocs.io/en/master/user/advanced/#link-headers
+        :type dict
+        :param access_token: an optional, pregenerated access token. If provided, the class instance will not
+        automatically try to request a new access token.
+        :type: access_token: str
+        :param kwargs:
+        """
+        super(DirectAccessV2, self).__init__(api_key, retries, backoff_factor, **kwargs)
+        self.client_id = client_id
+        self.client_secret = client_secret
+        self.links = links
+        self.access_token = access_token
+        self.url = self.url + "/v2/direct-access"
+        self.session.headers.update({"X-API-KEY": self.api_key})
+        self.session.hooks["response"].append(self._check_response)
+
+        if self.access_token:
+            self.session.headers["Authorization"] = "bearer {}".format(
+                self.access_token
+            )
+        else:
+            self.access_token = self.get_access_token()["access_token"]
+
+    def get_access_token(self):
+        """
+        Get an access token from /tokens endpoint. Automatically sets the Authorization header on the class instance's
+        session. Raises DAAuthException on error
+
+        :return: token response as dict
+        """
+        url = self.url + "/tokens"
+        if not self.api_key or not self.client_id or not self.client_secret:
+            raise DAAuthException(
+                "API_KEY, CLIENT_ID and CLIENT_SECRET are required to generate an access token"
+            )
+        self.session.headers["Authorization"] = "Basic {}".format(
+            base64.b64encode(
+                ":".join([self.client_id, self.client_secret]).encode()
+            ).decode()
+        )
+        self.session.headers["Content-Type"] = "application/x-www-form-urlencoded"
+
+        payload = {"grant_type": "client_credentials"}
+        response = self.session.post(url, params=payload)
+        self.logger.debug("Token response: " + json.dumps(response.json(), indent=2))
+        self.access_token = response.json()["access_token"]
+        self.session.headers["Authorization"] = "bearer {}".format(self.access_token)
+        return response.json()
+
+
+class DeveloperAPIv3(BaseAPI):
+    """Client for Enverus' Developer API Version 3"""
+
+    def __init__(
+            self,
+            api_key,
```

Comment:
I think will be a good idea to rename api_key to secret_key.
In our docs (Getting Started with Version 3),  we are using the Secret Key definition. It can make a little misunderstanding for users.